### PR TITLE
Add iron golem spawn egg

### PR DESCRIPTION
**Pull request by TheGallium**
Referring to issue #108

### CODE REVIEW

File: misc-eggs.sk [hunk 1]
```diff
@@ -183,6 +183,10 @@ the wild update:
 	(warden [spawn] egg|spawn warden)¦s = minecraft:warden_spawn_egg
 	[any] spawn egg¦s = any spawn egg, spawn allay, spawn frog, spawn tadpole, spawn warden
 
+the wild update after iron golem:
+	minecraft version = 1.19.3 or newer
+	([iron] golem [spawn] egg|spawn [iron] golem)¦s = minecraft:spawn_egg {Damage:50}
```
Comment (sovdeeth) [inline misc-eggs.sk]:
```suggestion
	([iron] golem [spawn] egg|spawn [iron] golem)¦s = minecraft:iron_golem_spawn_egg
```

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: misc-eggs.sk (hunk 1 above)
Comment (sovdeeth) [reply misc-eggs.sk]: You'll also need to add this to the all spawn eggs alias.

**Comment by Fusezion** [comment]
I'm not sure how I feel about making `iron` optional. as snow golem also exists and if we're missing iron we're likely missing that one

*and let's not talk about `spawn (entity)` even being valid across spawn eggs, this is just innately wrong
image

**Comment by sovdeeth** [comment]
> I'm not sure how I feel about making `iron` optional. as snow golem also exists and if we're missing iron we're likely missing that one
> 
> *and let's not talk about `spawn (entity)` even being valid across spawn eggs, this is just innately wrong image

good catch with the iron thing
as far as the second point, i agree, but for a different pr.

**Comment by TheGallium** [comment]
The `minecraft:iron_golem_spawn_egg` item was added in 1.19.3, using it in 1.19 will not work.
Also, for the making `iron` optional, I think `golem` should be the default but for the snow golem, `snow` would be required.

**Comment by sovdeeth** [comment]
I would prefer both be required. It's better to be explicit about thing to avoid confusion and future conflicts.

**Review by sovdeeth — APPROVED** [review]

**Review by Fusezion — APPROVED** [review]
At work but from what I can see everything that concerned me is fixed

**Review by TheBentoBox — DISMISSED** [review]

File: misc-eggs.sk [hunk 2]
```diff
@@ -183,6 +183,11 @@ the wild update:
 	(warden [spawn] egg|spawn warden)¦s = minecraft:warden_spawn_egg
 	[any] spawn egg¦s = any spawn egg, spawn allay, spawn frog, spawn tadpole, spawn warden
 
+the wild update after golems egg:
+	minecraft version = 1.19.3 or newer
+	[iron golem [spawn] egg|spawn iron golem)¦s = minecraft:iron_golem_spawn_egg
+	[snow golem [spawn] egg|spawn snow golem)¦s = minecraft:snow_golem_spawn_egg
```
Comment (TheBentoBox) [inline misc-eggs.sk]: Accidentally submitted my approval, but this is wrong -- the first square brace should be the open parentheses. Was this tested?

```suggestion
	(iron golem [spawn] egg|spawn iron golem)¦s = minecraft:iron_golem_spawn_egg
	(snow golem [spawn] egg|spawn snow golem)¦s = minecraft:snow_golem_spawn_egg
```

**Review by TheBentoBox — APPROVED** [review]

**Comment by ghost** [comment]
what about ender dragon & wither spawn eggs

File: misc-eggs.sk [hunk 3]
```diff
@@ -183,6 +183,11 @@ the wild update:
 	(warden [spawn] egg|spawn warden)¦s = minecraft:warden_spawn_egg
 	[any] spawn egg¦s = any spawn egg, spawn allay, spawn frog, spawn tadpole, spawn warden
 
+the wild update after golems egg:
+	minecraft version = 1.19.3 or newer
+	(iron golem [spawn] egg|spawn iron golem)¦s = minecraft:iron_golem_spawn_egg
+	(snow golem [spawn] egg|spawn snow golem)¦s = minecraft:snow_golem_spawn_egg
```
Comment (APickledWalrus) [inline misc-eggs.sk:189]:
```suggestion
	(snow golem [spawn] egg|spawn snow golem)¦s = minecraft:snow_golem_spawn_egg
	[any] spawn egg¦s = any spawn egg, spawn iron golem, spawn snow golem
```

**Review by APickledWalrus — APPROVED** [review]

**Comment by APickledWalrus** [comment]
Thanks for this!

**Pull request merged**